Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -229,7 +229,7 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
         cell.connectingForwards = indexPath.item != 0
 
         guard let url = urlString, !item.url.isAboutHomeURL else {
-            cell.site = Site(url: item.url.absoluteString, title: Strings.FirefoxHomePage)
+            cell.site = Site(url: item.url.absoluteString, title: Strings.New_Tab)
```

Comment:
Rename `New_Tab` text to "Home"

per: https://github.com/brave/brave-ios/issues/353